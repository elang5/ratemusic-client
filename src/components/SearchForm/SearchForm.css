.search-form {
  background: url('https://images.unsplash.com/3/www.madebyvadim.com.jpg?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1661&q=80') center;
  background-size: cover;
  background-repeat: no-repeat;
  padding: 4rem 2rem;
  text-align: right;
}


#album_search {
  font-size: 18px;
}

button {
  padding: 9px 13px;
  border-radius: 3px;
  box-shadow: 4px #1f1f21;
}

.button-container {
  text-align: right;
  padding-top: .5em;
}

#album_search {
  margin: .5rem 0;
}
