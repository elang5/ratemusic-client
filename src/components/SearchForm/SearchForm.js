import React, { Component } from 'react'
import './SearchForm.css'

export default class SearchForm extends Component {
  static defaultProps = {
    name: '',
    albums: [],
    handleSubmit: () => {}
  }

  constructor() {
    super()
    this.state = {
      album: '',
      albums: [],
      error: null
    }
  }

  componentDidMount() {
    this.setState({
      albums: this.props.albums
    })
  }


  render() {
    return (
      <header className="search-header">
      <form 
        className="search-form" 
        onSubmit={(e) => this.props.searchAlbums(e)}
      >
        <h1 className="red">Rate Music</h1>
        <div className="input-fields">
          <label htmlFor="album_search" className="search-label">
            {this.props.name}
          </label><br />
          <input type="text" id="album_search" name="album_search" />
        </div>
        <div className="button-container">
          <button type="submit">
            Search
          </button>
        </div>
      </form>
      </header>
    )
  }
}
