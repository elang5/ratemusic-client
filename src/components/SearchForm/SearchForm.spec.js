import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchForm from './SearchForm'

describe(`SearchForm Component`, () => {

  it(`renders the SearchForm component`, () => {
    const props = {
      name: 'Test',
      albums: []
    }
    const wrapper = shallow(<SearchForm {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})