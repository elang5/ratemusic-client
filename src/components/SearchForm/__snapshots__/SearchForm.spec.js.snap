// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Component renders the SearchForm component 1`] = `
<header
  className="search-header"
>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <h1
      className="red"
    >
      Rate Music
    </h1>
    <div
      className="input-fields"
    >
      <label
        className="search-label"
        htmlFor="album_search"
      >
        Test
      </label>
      <br />
      <input
        id="album_search"
        name="album_search"
        type="text"
      />
    </div>
    <div
      className="button-container"
    >
      <button
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</header>
`;
